lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -15,16 +16,71 @@ class FirebaseAppDistributionAction < Action
       FIREBASECMD_ACTION = "appdistribution:distribute".freeze
       BASE_URL = "https://firebaseappdistribution.googleapis.com"
       PATH = "/v1alpha/apps/"
+      MAX_POLLING_RETRIES = 60
+      POLLING_INTERVAL_S = 2
 
       extend Helper::FirebaseAppDistributionHelper
 
       def self.run(params)
         params.values # to validate all inputs before looking for the ipa/apk
-        validate_app!(params[:app])
+        app_id = nil
+        platform = Actions.lane_context[Actions::SharedValues::PLATFORM_NAME]
+        binary_path = Shellwords.escape(params[:ipa_path] || params[:apk_path])
+
+        if params[:app] # Set app_id if it is specified as a parameter
+          app_id = params[:app]
+        else
+          if platform == :ios
+            archive_path = Actions.lane_context[SharedValues::XCODEBUILD_ARCHIVE]
+            if archive_patharchive_path
+              app_id = findout_ios_app_id_from_archive(archive_path)
+            end
+          end
+          if app_id.nil?
+            app_id = UI.input("Your app's Firebase App ID. You can find the App ID in the Firebase console, on the General Settings page: ")
```

Comment:
Do we need to do this? Based on the comment above, my understanding was that the call to `params.values` checked the parameter values and if the app id is blank, prompted the user with the same message.

Reply:
I tried running the action without an app id and params.values does not ask me to input an app id. Instead we just get our error handling.

![Screen Shot 2020-06-22 at 5 11 43 PM](https://user-images.githubusercontent.com/32231345/85335881-452b9180-b4cd-11ea-95b6-6a1c0fc9beab.png)

Reply:
https://github.com/fastlane/fastlane-plugin-firebase_app_distribution/blob/d0a48bf604920f07c622eb457ecd484cc7409100/lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb#L129-L133 

Because the :app parameter is optional, the user won't be prompted to enter in the ID. If we set optional to False it will prompt the user if they don't specify it as a parameter but we decided to keep it optional because the App ID is optional if the platform is ios

Reply:
👍